packages/docs/content/dapp-kit/migration.mdx

```diff
@@ -194,17 +194,17 @@ export function ExampleComponent({ transaction }: { transaction: Transaction })
 ### 6. Replace Data Fetching Patterns
 
 React Query hooks are no longer available, giving you flexibility to choose your preferred data
-fetching solution. Use standard React patterns or integrate with TanStack Query if desired.
+fetching solution. The new client uses gRPC for communication with Sui nodes. Use standard React
+patterns or integrate with TanStack Query if desired.
 
 ```diff
 - import { useSuiClientQuery } from '@mysten/dapp-kit';
 + import { useCurrentClient } from '@mysten/dapp-kit-react';
 + import { useState, useEffect } from 'react';
-+ import type { SuiObjectData } from '@mysten/sui/client';
 
   export function ExampleComponent({ objectId }: { objectId: string }) {
 + 	const client = useCurrentClient();
-+ 	const [data, setData] = useState<SuiObjectData | null>(null);
++ 	const [data, setData] = useState<{ objectId?: string; contents?: { value?: Uint8Array } } | null>(null);
```

Comment:
We should probably not advise people do data fetching this way since really you should use a library